packages/router-api/src/client.rs

```diff
@@ -1,23 +1,34 @@
+use std::marker::PhantomData;
+
 use axelar_wasm_std::vec::VecExt;
-use cosmwasm_std::{to_json_binary, Addr, WasmMsg};
+use cosmwasm_std::{to_json_binary, Addr, CosmosMsg, Empty, WasmMsg};
 
 use crate::msg::ExecuteMsg;
 use crate::Message;
 
-pub struct Router {
+pub struct Router<T = Empty> {
```

Comment:
why does the router need to know about CustomMsg?

Reply:
1. This router client isn't using our client::ContractClient which IMO is bad. 2. The contracts using the router may need to return `CosmosMsg<SomeCustomMsg>`, for example the nexus gateway.

Reply:
can you follow up on that and reimplement it in terms of the contract client then in a separate PR?